frontend/src/metabase/writeback/components/ActionCreator/FormCreator/FormCreator.tsx

```diff
@@ -1,38 +1,92 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { t } from "ttag";
 
-import Icon from "metabase/components/Icon";
-
 import type { TemplateTag } from "metabase-types/types/Query";
+import type {
+  ActionFormSettings,
+  FieldSettings,
+} from "metabase/writeback/types";
+
+import { FieldSettingsPopover } from "./FieldSettingsPopover";
+import { getDefaultFormSettings, getDefaultFieldSettings } from "./utils";
+import { FormField } from "./FormField";
 
 import {
   FormItemWrapper,
   FormCreatorWrapper,
   FormItemName,
-  Input,
   FormSettings,
   EmptyFormPlaceholderWrapper,
 } from "./FormCreator.styled";
 
-export function FormCreator({ tags }: { tags: TemplateTag[] }) {
+export function FormCreator({
+  tags,
+  formSettings: passedFormSettings,
+  onChange,
+}: {
+  tags: TemplateTag[];
+  formSettings?: ActionFormSettings;
+  onChange: (formSettings: ActionFormSettings) => void;
+}) {
+  const [formSettings, setFormSettings] = useState<ActionFormSettings>(
+    passedFormSettings?.fields ? passedFormSettings : getDefaultFormSettings(),
+  );
+
+  useEffect(() => {
+    onChange(formSettings);
+  }, [formSettings, onChange]);
+
+  const handleChangeFieldSettings = (
+    tagId: string,
+    newFieldSettings: FieldSettings,
+  ) => {
+    setFormSettings({
+      ...formSettings,
+      fields: {
+        ...formSettings.fields,
+        [tagId]: newFieldSettings,
```

Comment:
This update pattern leaves open the possibility of orphaned field settings for parameters that are removed. 🤔  This won't hurt anything, but we could end up storing useless data.

I think a cleanup function before save is probably the best way to handle this? How do we clean up orphaned parameters elsewhere?

Reply:
added a cleanup function in https://github.com/metabase/metabase/pull/25119/commits/effde5f0499c0c719aab5d528f276c802dedc581